plugins/destination/s3/client/write.go

```diff
@@ -132,14 +134,12 @@ func (c *Client) MigrateTable(ctx context.Context, ch <-chan *message.WriteMigra
 		c.initializedTables[table.Name] = objKey
 		s, err := c.createObject(ctx, table, objKey)
 		if err != nil {
-			return err
-		}
-		err = s.Finish()
-		if err != nil {
-			return err
+			errs = errors.Join(errs, err)
```

Comment:
Not returning early is not perfect but the current contract with StreamingBatchWriter requires this channel to be consumed to completion.